src/database/initialization/user_to_json.sql

```diff
@@ -2,9 +2,57 @@ SELECT drop_function('user_to_json');
 
 CREATE FUNCTION user_to_json("user" users)
 RETURNS JSONB AS $$
+DECLARE
+  roles_json JSONB := NULL::JSONB;
+  changemaker_roles_json JSONB := NULL::JSONB;
+  funder_roles_json JSONB := NULL::JSONB;
+  data_provider_roles_json JSONB := NULL::JSONB;
 BEGIN
+  changemaker_roles_json := (
+    SELECT jsonb_object_agg(
+      changemaker_role_maps.changemaker_id, changemaker_role_maps.role_map
+    )
+    FROM (
+      SELECT changemaker_roles.changemaker_id, jsonb_object_agg(changemaker_roles.access_type, TRUE) AS role_map
```

Comment:
5. What does `TRUE` do here? (I can investigate later but want to finish this review quickly).

Reply:
The result of this is an object where keys are access types, and values are booleans (well, `true` specifically).

So:

`{ edit: true }`

A client can then say something like `if (user.roles.funders.fordFoundation.edit) then ...`

That said, this might not be the right object structure.  Maybe an array of access types would be better.

Reply:
On reflection I think an array is more honest; the fact that `false` values don't get entries in the object makes the object misleading.  An array naturally communicates that it is only communicating what is present.

Updating this (and `TRUE` will no longer be there)

Reply:
Oh, nice. I truly was asking and didn't mean to say "do a bunch more" but I'm glad that the question raised a better alternative!